src/sacn/sockets.c

```diff
@@ -555,16 +639,77 @@ etcpal_error_t sacn_validate_netint_config(SacnMcastInterface* netints, size_t n
   }
 }
 
+#if SACN_DYNAMIC_MEM
+etcpal_error_t sacn_initialize_internal_netints(EtcPalMcastNetintId** internal_netints, size_t* num_internal_netints,
+                                                SacnMcastInterface* app_netints, size_t num_app_netints)
+#else
+etcpal_error_t sacn_initialize_internal_netints(EtcPalMcastNetintId (*internal_netints)[SACN_MAX_NETINTS],
+                                                size_t* num_internal_netints, SacnMcastInterface* app_netints,
+                                                size_t num_app_netints)
+#endif
+{
+  size_t num_valid_netints = 0u;
+  etcpal_error_t result = sacn_validate_netint_config(app_netints, num_app_netints, &num_valid_netints);
+
+  if (result == kEtcPalErrOk)
+  {
+    if (app_netints)
+    {
+#if SACN_DYNAMIC_MEM
+      *internal_netints = calloc(num_valid_netints, sizeof(EtcPalMcastNetintId));
+
+      if (!(*internal_netints))
+        result = kEtcPalErrNoMem;
+#else
+      if (num_app_netints > SACN_MAX_NETINTS)
+        result = kEtcPalErrNoMem;
+#endif
+
+      if (result == kEtcPalErrOk)
+      {
+        for (size_t read_index = 0u, write_index = 0u; read_index < num_app_netints; ++read_index)
+        {
+          if (app_netints[read_index].status == kEtcPalErrOk)
+          {
+            memcpy(*internal_netints + write_index, &app_netints[read_index].iface, sizeof(EtcPalMcastNetintId));
```

Comment:
Unless I'm missing something, you can just do:
```
internal_netints[write_index] = app_netints[read_index].iface;
```

Reply:
Resolved in https://github.com/ETCLabs/sACN/pull/32/commits/dcd53cdc3327dd8a8d4e6e8ef664245f344a88d4.